c7n/resources/efs.py

```diff
@@ -193,3 +194,42 @@ def process(self, resources):
                     LifecyclePolicies=op_map.get(self.data.get('state')))
             except client.exceptions.FileSystemNotFound:
                 continue
+
+
+@ElasticFileSystem.filter_registry.register('lifecycle-policy')
+class LifecyclePolicy(Filter):
+    """Filters efs based on the state of lifecycle policies
+
+    :example:
+
+      .. code-block:: yaml
+
+            policies:
+              - name: efs-filter-lifecycle
+                resource: efs
+                filters:
+                  - type: lifecycle-policy
+                    state: true
+
+    """
+    schema = type_schema(
+        'lifecycle-policy',
+        state={'type': 'boolean'},
+        required=['state'])
+
+    permissions = ('elasticfilesystem:DescribeLifecycleConfiguration',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('efs')
+        results = []
+        for r in resources:
+            try:
+                response = client.describe_lifecycle_configuration(
+                    FileSystemId=r['FileSystemId'])
+            except client.exceptions.FileSystemNotFound:
+                continue
+            if self.data.get('state') and response.get('LifecyclePolicies'):
```

Comment:
I feel like we should go ahead add in the ability to check on the actual policy content with this filter, ie is it 7 days, 14 etc. The lifecycle policies are an array but only support a singleton value at the moment, so from a filter perspective perhaps ie `value: AFTER_7` with `state: present` to only find those configured for 7 days.

Reply:
Cool. Was thinking more of making `value` an optional parameter that would be of type array.  Although, as you said, it only supports a single value right now but might be worth keeping it as an array for future. Thoughts? Thanks

Reply:
I think we can do it as singleton non array, and if they expose more options later we can extend it w/ compatibility.

Reply:
Cool, thanks! Also realized we can manipulate `value` to attain any possible filter. So I made it a mandatory field to keep things simple.